needle.pm

```diff
@@ -72,21 +73,21 @@ sub new {
 
     my $gotmatch;
     for my $area (@{$json->{area}}) {
-        my $a = {};
+        my $hash = {};
```

Comment:
When cleaning up anyways, maybe rename `$area` to `$area_from_json` or `$raw_area` and `$hash`/`$a` to `$area`. Same for the loop in `sub save`.

Reply:
I'm not really cleaning up here, `$a` and `$b` are built-in Perl variables with special meaning. Using them outside their intended purpose is a bug.